pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/heal.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022 Doc.ai and/or its affiliates.
+//
+// Copyright (c) 2021-2022 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package iprule
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+
+	link "github.com/networkservicemesh/sdk-kernel/pkg/kernel"
+)
+
+func recoverTableIDs(ctx context.Context, conn *networkservice.Connection, tableIDs *Map) error {
+	if mechanism := kernel.ToMechanism(conn.GetMechanism()); mechanism != nil && mechanism.GetVLAN() == 0 {
+		netlinkHandle, err := link.GetNetlinkHandle(mechanism.GetNetNSURL())
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		defer netlinkHandle.Close()
+
+		podRules, err := netlinkHandle.RuleList(netlink.FAMILY_ALL)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		ps, ok := tableIDs.Load(conn.GetId())
+		if !ok {
+			if len(conn.Context.IpContext.Policies) == 0 {
+				return nil
+			}
+			ps = make(map[int]*networkservice.PolicyRoute)
+			tableIDs.Store(conn.GetId(), ps)
+		}
```

Comment:
We will be here only once for each client, so if the map is not empty, then nothing needs to be restored.
```
else {
    return nil
}
```